### PR TITLE
Fix CMake policy warnings

### PR DESCRIPTION
**Pull request by JJL772**
Fix CMake policy warnings and upgrade CMake minimum version.

### Description

* Fix CMake policy [CMP0148](https://cmake.org/cmake/help/latest/policy/CMP0148.html) by upgrading to `FindPython3` from the old find package methods for Python
* Fix CMake policy [CMP0074](https://cmake.org/cmake/help/latest/policy/CMP0074.html#policy:CMP0074) by correcting the case of the `Boost_ROOT` variable. Ironically they re-added the all-caps functionality in CMake 3.27.
* Increased minimum CMake version to 3.15

### CODE REVIEW

File: buildroot-2019.08-x86_64.cmake [hunk 1]
```diff
@@ -24,8 +24,8 @@ set(ZeroMQ_LIBRARY     /afs/slac/g/lcls/package/libzmq/zeromq-4.3.4/buildroot-20
 set(ZeroMQ_INCLUDE_DIR /afs/slac/g/lcls/package/libzmq/zeromq-4.3.4/buildroot-2019.08-x86_64/include)
 
 # Define the location of python3 (cross-compiled)
-set(PYTHON_LIBRARY     /afs/slac/g/lcls/package/python/3.6.1/buildroot-2019.08-x86_64/lib/libpython3.6m.so)
-set(PYTHON_INCLUDE_DIR /afs/slac/g/lcls/package/python/3.6.1/buildroot-2019.08-x86_64/include/python3.6m)
+set(Python3_LIBRARY     /afs/slac/g/lcls/package/python/3.6.1/buildroot-2019.08-x86_64/lib/libpython3.6m.so)
```
[the remark this answers is not in the document]
Comment (JJL772) [reply buildroot-2019.08-x86_64.cmake:27]: Our `$PACKAGE_TOP` in s3df land is now `/sdf/sw/epics/package`

I think I can change this toolchain to use `$PACKAGE_TOP` so it's portable.

[the remark this answers is not in the document]
Comment (ruck314) [reply buildroot-2019.08-x86_64.cmake:27]: @JJL772 We should remove all the AFS path from the cmake at this point and change them to SDF

**Comment by JJL772** [comment]
Going to mark this as a draft until I fix the buildroot cross compile

**Comment by slacrherbst** [comment]
I was able to verify this works on Ubuntu 20.04, Ubuntu 22.04, Rocky 8, Rocky 9, archlinux and anaconda. From my view this is good to go.

**Review by slacrherbst — APPROVED** [review]

**Comment by JJL772** [comment]
Thanks for testing it!

**Pull request merged**